kotest-assertions/kotest-assertions-json/src/commonMain/kotlin/io/kotest/assertions/json/JsonMatchers.kt

```diff
@@ -46,25 +46,61 @@ fun matchJson(expected: String?) = object : Matcher<String?> {
          return MatcherResult(
             false,
             { "expected: expected json to be valid json: $expected" },
-            {
-               "expected: expected json to be invalid json: $expected"
-            })
+            { "expected: expected json to be invalid json: $expected" }
+         )
       }
 
       return ComparableMatcherResult(
          actualJson == expectedJson,
-         {
-            "expected json to match, but they differed\n\n"
-         },
-         {
-            "expected not to match with: $expectedJson but match: $actualJson"
-         },
+         { "expected json to match, but they differed\n\n" },
+         { "expected not to match with: $expectedJson but match: $actualJson" },
          actualJson.toString(),
          expectedJson.toString()
       )
    }
 }
 
+@OptIn(ExperimentalSerializationApi::class)
+fun beValidJson() = object : Matcher<String?> {
+   override fun test(value: String?): MatcherResult {
+      return try {
+         value?.let(pretty::parseToJsonElement)
+         MatcherResult(
+              true,
+              { "expected: actual json to be valid json: $value" },
+              { "expected: actual json to be invalid json: $value" }
+         )
+      } catch (ex: Exception) {
+         MatcherResult(
+              false,
+              { "expected: actual json to be valid json: $value" },
+              { "expected: actual json to be invalid json: $value" }
+         )
+      }
+   }
+}
+
+@OptIn(ExperimentalSerializationApi::class)
+fun beElementOf(kClass: KClass<*>) = object : Matcher<String?> {
```

Comment:
maybe `beJsonType` ?

Reply:
Ill rename when I update the docs. :+1

Reply:
Ok will merge and then release 5.2 this weekend